src/ZPublisher/tests/testHTTPRequest.py

```diff
@@ -766,6 +766,18 @@ def test_processInputs_w_cookie_parsing(self):
         self.assertEqual(req.cookies['multi2'],
                          'cookie data with unquoted spaces')
 
+    def test_processInputs_xmlrpc(self):
+        TEST_METHOD_CALL = (
+            b'<?xml version="1.0"?>'
+            b'<methodCall><methodName>test</methodName></methodCall>'
+        )
+        environ = self._makePostEnviron(body=TEST_METHOD_CALL)
+        environ['CONTENT_TYPE'] = 'text/xml'
+        req = self._makeOne(stdin=BytesIO(TEST_METHOD_CALL), environ=environ)
+        req.processInputs()
+        self.assertEqual(req.PATH_INFO, '/test')
+        self.assertEqual(req.args, ())
```

Comment:
Exercise the `qs_on_post` bit?

Reply:
Thanks for asking about this -- helped me realize that I needed a completely different approach (and additional test) to avoid losing the query string.